tests/test_s3boto3.py

```diff
@@ -756,6 +756,12 @@ def test_override_settings(self):
             storage = s3boto3.S3Boto3Storage()
             self.assertEqual(storage.location, 'foo2')
 
+    def test_s3_uri_scheme_override(self):
+        with override_settings(AWS_S3_USE_S3_URI_SCHEME=True):
+            storage = s3boto3.S3Boto3Storage()
+            self.assertEqual(storage.use_s3_uri_scheme, 's3')
+            self.assertEqual(storage.url_protocol, 's3')
```

Comment:
Since is it required for custom domain to be enabled we should add a test that constructs a connection and validate that it is of the correct format in all cases.